File: native/packages/react-native-backpack/libraries/Panel/android/react/src/main/java/net/skyscaner/backpack/panel/PanelManager.java
package net.skyscaner.backpack.panel;

import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;

import net.skyscanner.backpack.panel.BpkPanel;


public class PanelManager extends SimpleViewManager<BpkPanel> {

    public static final String REACT_CLASS = "BPKPanel";

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected BpkPanel createViewInstance(ThemedReactContext reactContext) {
        return new BpkPanel(reactContext);
    }
}

File: native/packages/react-native-backpack/libraries/Panel/android/react/src/main/java/net/skyscaner/backpack/panel/PanelManager.kt
package net.skyscanner.backpack

import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.ViewGroupManager
import com.facebook.react.uimanager.annotations.ReactProp


class PanelManager : ViewGroupManager<BpkPanel>() {

    override fun getName(): String {
        return REACT_CLASS
    }

    override fun createViewInstance(reactContext: ThemedReactContext): BpkPanel {
        return BpkPanel(reactContext)
    }

    @ReactProp(name = "padded", defaultBoolean = true)
    fun setPadding(panel: BpkPanel, padded: Boolean) {
        panel.mPadding = padded
    }

    companion object {
        val REACT_CLASS = "BPKPanelWrapper"
    }

}
